### PR TITLE
mcomix: 2.2.1 -> 3.0.0

### PR DESCRIPTION
**Pull request by thiagokokada**
## Description of changes

- Migrate to by-name
- Bump: 2.2.1 -> 3.0.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/mc/mcomix/package.nix [hunk 1]
```diff
@@ -8,37 +8,53 @@
...
 
-  # prevent double wrapping
+  # Prevent double wrapping
   dontWrapGApps = true;
 
   preFixup = ''
     makeWrapperArgs+=(
       "''${gappsWrapperArgs[@]}"
-      "--prefix" "PATH" ":" "${lib.makeBinPath ([ p7zip lhasa mupdf ] ++ lib.optional (unrarSupport) unrar)}"
+      "--prefix" "PATH" ":" "${lib.makeBinPath ([ p7zip ] ++ lib.optional unrarSupport unrar)}"
```
Comment (AndersonTorres) [inline pkgs/by-name/mc/mcomix/package.nix:57]: was those removed?

File: pkgs/by-name/mc/mcomix/package.nix [hunk 2]
```diff
@@ -8,37 +8,53 @@
...
 }:
 
 python3.pkgs.buildPythonApplication rec {
   pname = "mcomix";
-  version = "2.2.1";
+  version = "3.0.0";
+  pyproject = true;
 
   src = fetchurl {
     url = "mirror://sourceforge/mcomix/${pname}-${version}.tar.gz";
```
Comment (AndersonTorres) [inline pkgs/by-name/mc/mcomix/package.nix:25]: why not

```suggestion
    url = "mirror://sourceforge/mcomix/mcomix-${version}.tar.gz";
```

**Review by AndersonTorres — COMMENTED** [review]
just suggestions

File: pkgs/by-name/mc/mcomix/package.nix (hunk 1 above)
Comment (thiagokokada) [reply pkgs/by-name/mc/mcomix/package.nix:57]: - `mupdf` was replaced with `pymupdf`
- if `p7zip` is installed it is preferred over `lhasa`: https://sourceforge.net/p/mcomix/git/ci/master/tree/mcomix/archive_tools.py. So doesn't make sense to include both

File: pkgs/by-name/mc/mcomix/package.nix (hunk 2 above)
Comment (thiagokokada) [reply pkgs/by-name/mc/mcomix/package.nix]: Done.

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/by-name/mc/mcomix/package.nix [hunk 3]
```diff
@@ -8,37 +8,53 @@
...
 python3.pkgs.buildPythonApplication rec {
   pname = "mcomix";
-  version = "2.2.1";
+  version = "3.0.0";
+  pyproject = true;
 
   src = fetchurl {
-    url = "mirror://sourceforge/mcomix/${pname}-${version}.tar.gz";
-    hash = "sha256-fmnlPhNCN6YR3lW2YCMEAbEiWVigcfFDq1tDQ1eTNkA=";
+    url = "mirror://sourceforge/mcomix/mcomix-${version}.tar.gz";
```
[the remark this answers is not in the document]
Comment (thiagokokada) [reply pkgs/by-name/mc/mcomix/package.nix:25]: https://github.com/NixOS/nixpkgs/pull/279598#discussion_r1444620777

This is really something that depends on the person, but in general it is arguably easier to read an URL that does not parametrize unnecessary things like `pname`.

[the remark this answers is not in the document]
Comment (AndersonTorres) [reply pkgs/by-name/mc/mcomix/package.nix:25]: Also

https://github.com/NixOS/nixpkgs/pull/278611

**Comment by lucasew** [comment]
Result of `nixpkgs-review pr 279598` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>mcomix</li>
    <li>mcomix.dist</li>
  </ul>
</details>

**Pull request merged**